src/main/java/org/example/springbootintro/model/CartItem.java

```diff
@@ -0,0 +1,44 @@
+package org.example.springbootintro.model;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.Table;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+
+@Entity
+@Getter
+@Setter
+@Table(name = "cart_items")
+@SQLDelete(sql = "UPDATE cart_items SET is_deleted = true WHERE id =?")
+@SQLRestriction(value = "is_deleted=false")
```

Comment:
wouldn't say that we need a soft delete for cart items, does not seem like a very important an irreplaceable information for me, and user can update their cart multiple times, remove some books from it several times a day, should we store all that? I'd say no